common/include/pcl/PCLHeader.h

```diff
@@ -43,6 +43,12 @@ namespace pcl
     return (out);
   }
 
+  inline bool operator== (const PCLHeader &lhs, const PCLHeader &rhs)
+  {
+    return (&lhs == &rhs) ||
+      (lhs.seq == rhs.seq && lhs.stamp == rhs.stamp && lhs.frame_id == rhs.frame_id);
+  }
+
 } // namespace pcl
```

Comment:
It is implicitly used on the test [here](https://github.com/SergioRAgostinho/pcl/blob/sift_viewpoint/test/keypoints/test_keypoints.cpp#L84), and that is what forced me to define it. I thought that it was probably safe to write the == overload than to make the check on each individual member of the struct on the test. That way, if the struct changes in the future, the editor will know that he needs to update the operator as well, and test will still be valid.